### ISSUE TITLE
Permissions on files in ubuntu packages

### ISSUE THREAD

**Issue by candlerb** [zatosource/zato#379]
Files in the Ubuntu packages are owned by, and writable by, the zato user; furtherore, some files are not world-readable. e.g.

```
$ ls -l /opt/zato/2.0.1/eggs/blist-1.3.6-py2.7-linux-x86_64.egg/EGG-INFO/top_level.txt
-rw-r----- 1 zato zato 6 Feb  2 17:13 /opt/zato/2.0.1/eggs/blist-1.3.6-py2.7-linux-x86_64.egg/EGG-INFO/top_level.txt
```

Consequences:
- You cannot run tests except as the zato user (you get "permission denied" when it tries to read the file shown above)
- If a service running as the zato user were exploited, these files could be changed

I suggest that they be owned by root:root and mode 644 (or 755 for executables)

**Comment by dsuch** [comment]
Cheers - looking into it.

**Comment by dsuch** [comment]
I'm discussing it with @andy-wr who is the mastermind behind all the OS packages, will keep you updated. Thanks.

**Comment by candlerb** [comment]
FYI, still exists in 2.0.3

~~~
$ find /opt/zato/2.0.3 ! -perm /004 -print0 | xargs -0 ls -ld
-rw------- 1 zato zato  16286 Mar 26 14:27 /opt/zato/2.0.3/eggs/bcrypt-1.0.2-py2.7.egg/bcrypt/crypt_blowfish-1.2/crypt.3
-rw------- 1 zato zato    384 Mar 26 14:27 /opt/zato/2.0.3/eggs/bcrypt-1.0.2-py2.7.egg/bcrypt/crypt_blowfish-1.2/crypt_blowfish.h
-rw------- 1 zato zato   3979 Mar 26 14:27 /opt/zato/2.0.3/eggs/bcrypt-1.0.2-py2.7.egg/bcrypt/crypt_blowfish-1.2/crypt_gensalt.c
-rw------- 1 zato zato   1157 Mar 26 14:27 /opt/zato/2.0.3/eggs/bcrypt-1.0.2-py2.7.egg/bcrypt/crypt_blowfish-1.2/crypt_gensalt.h
-rw------- 1 zato zato    783 Mar 26 14:27 /opt/zato/2.0.3/eggs/bcrypt-1.0.2-py2.7.egg/bcrypt/crypt_blowfish-1.2/crypt.h
-rw------- 1 zato zato   1681 Mar 26 14:27 /opt/zato/2.0.3/eggs/bcrypt-1.0.2-py2.7.egg/bcrypt/crypt_blowfish-1.2/glibc-2.1.3-crypt.diff
-rw------- 1 zato zato   1767 Mar 26 14:27 /opt/zato/2.0.3/eggs/bcrypt-1.0.2-py2.7.egg/bcrypt/crypt_blowfish-1.2/glibc-2.14-crypt.diff
-rw------- 1 zato zato   1460 Mar 26 14:27 /opt/zato/2.0.3/eggs/bcrypt-1.0.2-py2.7.egg/bcrypt/crypt_blowfish-1.2/glibc-2.3.6-crypt.diff
-rw------- 1 zato zato    898 Mar 26 14:27 /opt/zato/2.0.3/eggs/bcrypt-1.0.2-py2.7.egg/bcrypt/crypt_blowfish-1.2/LINKS
-rw------- 1 zato zato   2072 Mar 26 14:27 /opt/zato/2.0.3/eggs/bcrypt-1.0.2-py2.7.egg/bcrypt/crypt_blowfish-1.2/Makefile
-rw------- 1 zato zato   1456 Mar 26 14:27 /opt/zato/2.0.3/eggs/bcrypt-1.0.2-py2.7.egg/bcrypt/crypt_blowfish-1.2/ow-crypt.h
-rw------- 1 zato zato   1304 Mar 26 14:27 /opt/zato/2.0.3/eggs/bcrypt-1.0.2-py2.7.egg/bcrypt/crypt_blowfish-1.2/PERFORMANCE
-rw------- 1 zato zato   3312 Mar 26 14:27 /opt/zato/2.0.3/eggs/bcrypt-1.0.2-py2.7.egg/bcrypt/crypt_blowfish-1.2/README
-rw------- 1 zato zato  15264 Mar 26 14:27 /opt/zato/2.0.3/eggs/bcrypt-1.0.2-py2.7.egg/bcrypt/crypt_blowfish-1.2/wrapper.c
-rw------- 1 zato zato   4221 Mar 26 14:27 /opt/zato/2.0.3/eggs/bcrypt-1.0.2-py2.7.egg/bcrypt/crypt_blowfish-1.2/x86.S
-rw-r----- 1 zato zato      1 Mar 26 14:27 /opt/zato/2.0.3/eggs/blist-1.3.6-py2.7-linux-x86_64.egg/EGG-INFO/dependency_links.txt
-rw-r----- 1 zato zato      1 Mar 26 14:27 /opt/zato/2.0.3/eggs/blist-1.3.6-py2.7-linux-x86_64.egg/EGG-INFO/not-zip-safe
-rw-r----- 1 zato zato   8790 Mar 26 14:27 /opt/zato/2.0.3/eggs/blist-1.3.6-py2.7-linux-x86_64.egg/EGG-INFO/PKG-INFO
-rw-r----- 1 zato zato    644 Mar 26 14:27 /opt/zato/2.0.3/eggs/blist-1.3.6-py2.7-linux-x86_64.egg/EGG-INFO/SOURCES.txt
-rw-r----- 1 zato zato      6 Mar 26 14:27 /opt/zato/2.0.3/eggs/blist-1.3.6-py2.7-linux-x86_64.egg/EGG-INFO/top_level.txt
-rw-r----- 1 zato zato      1 Mar 26 14:27 /opt/zato/2.0.3/eggs/httplib2-0.8-py2.7.egg/EGG-INFO/dependency_links.txt
-rw-r----- 1 zato zato   2337 Mar 26 14:27 /opt/zato/2.0.3/eggs/httplib2-0.8-py2.7.egg/EGG-INFO/PKG-INFO
-rw-r----- 1 zato zato    797 Mar 26 14:27 /opt/zato/2.0.3/eggs/httplib2-0.8-py2.7.egg/EGG-INFO/SOURCES.txt
-rw-r----- 1 zato zato      9 Mar 26 14:27 /opt/zato/2.0.3/eggs/httplib2-0.8-py2.7.egg/EGG-INFO/top_level.txt
-rw-r----- 1 zato zato  39670 Mar 26 14:27 /opt/zato/2.0.3/eggs/httplib2-0.8-py2.7.egg/httplib2/cacerts.txt
-rw-r----- 1 zato zato 202262 Mar 26 14:27 /opt/zato/2.0.3/eggs/Paste-1.7.5.1-py2.7.egg/paste/evalexception/media/MochiKit.packed.js
-rw------- 1 zato zato      1 Mar 26 14:27 /opt/zato/2.0.3/eggs/prettytable-0.7.2-py2.7.egg/EGG-INFO/dependency_links.txt
-rw------- 1 zato zato    710 Mar 26 14:27 /opt/zato/2.0.3/eggs/prettytable-0.7.2-py2.7.egg/EGG-INFO/PKG-INFO
-rw------- 1 zato zato    230 Mar 26 14:27 /opt/zato/2.0.3/eggs/prettytable-0.7.2-py2.7.egg/EGG-INFO/SOURCES.txt
-rw------- 1 zato zato     12 Mar 26 14:27 /opt/zato/2.0.3/eggs/prettytable-0.7.2-py2.7.egg/EGG-INFO/top_level.txt
-rw------- 1 zato zato 198578 Mar 26 14:27 /opt/zato/2.0.3/eggs/python_dateutil-2.2-py2.7.egg/dateutil/zoneinfo/zoneinfo--latest.tar.gz
-rw------- 1 zato zato      1 Mar 26 14:27 /opt/zato/2.0.3/eggs/python_dateutil-2.2-py2.7.egg/EGG-INFO/dependency_links.txt
-rw------- 1 zato zato      1 Mar 26 14:27 /opt/zato/2.0.3/eggs/python_dateutil-2.2-py2.7.egg/EGG-INFO/not-zip-safe
-rw------- 1 zato zato    970 Mar 26 14:27 /opt/zato/2.0.3/eggs/python_dateutil-2.2-py2.7.egg/EGG-INFO/PKG-INFO
-rw------- 1 zato zato      4 Mar 26 14:27 /opt/zato/2.0.3/eggs/python_dateutil-2.2-py2.7.egg/EGG-INFO/requires.txt
-rw------- 1 zato zato    563 Mar 26 14:27 /opt/zato/2.0.3/eggs/python_dateutil-2.2-py2.7.egg/EGG-INFO/SOURCES.txt
-rw------- 1 zato zato      9 Mar 26 14:27 /opt/zato/2.0.3/eggs/python_dateutil-2.2-py2.7.egg/EGG-INFO/top_level.txt
~~~

The above command lists all files which are not world-readable.

Also, eveything under /opt/zato/2.0.3 is owned by `zato:zato` rather than `root:root`.

**Comment by candlerb** [comment]
An example of where it currently fails:

~~~
PYTHONPATH=. /opt/zato/2.0.3/bin/alembic upgrade head
~~~

does not work unless it is run as either "zato" or "root"

**Comment by andy-wr** [comment]
OK, working on it.

**Comment by candlerb** [comment]
FYI, this is partly fixed in 2.0.6.  The files are all world-readable now, so can be used by non-zato users; however they are still owned by 'zato' and thus vulnerable to being overwritten by zato applications.

~~~
$ find /opt/zato/2.0.6 ! -perm /004 -print0 | xargs -0 ls -ld
drwxrwxr-x 23 zato zato 4096 Oct 22 16:45 .
$ find /opt/zato/2.0.6 ! -perm /040 -print0 | xargs -0 ls -ld
drwxrwxr-x 23 zato zato 4096 Oct 22 16:45 .
$ find /opt/zato/2.0.6 -type d ! -perm /001 -print0 | xargs -0 ls -ld
drwxrwxr-x 23 zato zato 4096 Oct 22 16:45 .
$ find /opt/zato/2.0.6 -type d ! -perm /010 -print0 | xargs -0 ls -ld
drwxrwxr-x 23 zato zato 4096 Oct 22 16:45 .
~~~
